File: ABC/ABC109/C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int gcd(int a, int b) {
  if (b == 0) {
    return a;
  }
  return gcd(b, a % b);
}

int main() {

  int N, X;
  cin >> N >> X;

  vector<int> x(N);
  for (int i = 0; i < N; i++) {
    cin >> x[i];
  }

  int answer = abs(x[0] - X);

  for (int i = 0; i < N; i++){
    int gcd_ = abs(x[i] - X);
    answer = gcd(gcd_, answer);
  }

  cout << answer << endl;

  return 0;

}

File: ABC/ABC109/C/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);

    int N = sc.nextInt();
    int X = sc.nextInt();

    int[] x = new int[N];

    for (int i = 0; i < N; i++) {
      x[i] = sc.nextInt();
    }

    sc.close();

    long g = Math.abs(x[0] - X);

    for (int i = 1; i < N; i++) {
       g = gcd(g, Math.abs(x[i] - X));
    }

    System.out.println(g);

  }

  public static long gcd(long x, long y){

    if (x < y) {
      long t = x;
      x = y;
      y = t;
    }

    while (y > 0) {
      long r = x % y;
      x = y;
      y = r;
    }

    return x;
    
  }

}